### PR TITLE
add logout navigation if user is logged in

### LINKED ISSUE #139: logout option should hide when user is not logged in cluster
Hide the logout option from cluster navigation actions if user is not logged in to cluster.

### PR DESCRIPTION
**Pull request by mohitsuman**
fix #139 

logout

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -252,7 +252,7 @@
         },
         {
           "command": "openshift.explorer.logout",
-          "when": "view == openshiftProjectExplorer",
+          "when": "view == openshiftProjectExplorer && isLoggedIn",
```
Comment (dgolovin) [inline package.json]: should be `context == "isLoggedIn"` otherwise it does not work.

File: src/extension.ts [hunk 2]
```diff
@@ -416,6 +416,7 @@ export namespace Openshift {
         const loginMessage = async (clusterURL, result, explorer)=> {
             if (result.stderr === "") {
                 explorer.refresh();
+                vscode.commands.executeCommand('setContext', 'isLoggedIn', true);
```
Comment (dgolovin) [inline src/extension.ts]: You set context only when you login/logout commands executed, but you are forgetting about initial state, right after extension activation.

File: src/cli.ts [hunk 3]
```diff
@@ -127,7 +127,7 @@ async function getToolLocation(cmd): Promise<string> {
         } catch (error) {
             const response = await vscode.window.showInformationMessage(
                 `Cannot find ${tools[cmd].description}.`, 'Download and install', 'Help', 'Cancel');
-            if(response === 'Download and install') {
+            if (response === 'Download and install') {
```
Comment (dgolovin) [inline src/cli.ts]: Please do not mix fix with formatting changes.

Comment (mohitsuman) [reply src/cli.ts]: sure. will keep this in mind :)

File: package.json (hunk 1 above)
Comment (mohitsuman) [reply package.json]: It works without setting the context word. Let me see if I can add a gif for it.

File: src/extension.ts (hunk 2 above)
Comment (mohitsuman) [reply src/extension.ts]: I have added the initial state scenario too in this:
https://github.com/redhat-developer/vscode-openshift-tools/pull/141/files#diff-f8d556788b6298850280fcb4c474b5e2R168

File: package.json (hunk 1 above)
Comment (mohitsuman) [reply package.json]: @dgolovin Updated the description with the gif.

File: src/extension.ts [hunk 4]
```diff
@@ -429,6 +430,7 @@ export namespace Openshift {
                     if (result.stderr === "") {
                         explorer.refresh();
                         const logoutInfo = await vscode.window.showInformationMessage(`Successfully logged out. Do you want to login to a new cluster`, 'Yes', 'No');
+                        vscode.commands.executeCommand('setContext', 'isLoggedIn', false);
```
Comment (jrichter1) [inline src/extension.ts]: Just a minor nitpick - please put this line before the info message so the log out option disappears along with the cluster.

Comment (mohitsuman) [reply src/extension.ts]: Fixed this.

**Pull request merged**